src/Sentry.GraphQl/GraphQlContentExtractor.cs

```diff
@@ -0,0 +1,90 @@
+namespace Sentry.GraphQl;
+
+/// <summary>
+/// Helper class to extract the content from GraphQL requests and responses
+/// </summary>
+internal class GraphQlContentExtractor
+{
+    private readonly SentryOptions? _options;
+
+    public GraphQlContentExtractor(SentryOptions? options)
+    {
+        _options = options;
+    }
+
+    /// <summary>
+    /// Extracts a <see cref="GraphQLRequest"/> from the <paramref name="request"/>
+    /// </summary>
+    /// <param name="request"></param>
+    /// <returns></returns>
+    public async Task<GraphQlRequestContent?> ExtractRequestContentAsync(HttpRequestMessage request)
+    {
+        var json = await ExtractContentAsync(request?.Content).ConfigureAwait(false);
+        return json is not null ? new GraphQlRequestContent(json, _options) : null;
+    }
+
+    /// <summary>
+    /// Extracts the Json text a <paramref name="response"/>
+    /// </summary>
+    /// <param name="response"></param>
+    /// <returns></returns>
+    public async Task<JsonElement?> ExtractResponseContentAsync(HttpResponseMessage response)
+    {
+        var json = await ExtractContentAsync(response?.Content).ConfigureAwait(false);
+        return (json is not null) ? JsonDocument.Parse(json).RootElement.Clone() : null;
+    }
+
+    void TrySeek(Stream? stream, long position)
+    {
+        if (stream?.CanSeek ?? false)
+        {
+            stream.Position = position;
+        }
+    }
+
+    public async Task<string?> ExtractContentAsync(HttpContent? content)
+    {
+        // Not to throw on code that ignores nullability warnings.
+        if (content is null)
+        {
+            return null;
+        }
+
+        Stream contentStream;
+        try
+        {
+            await content.LoadIntoBufferAsync().ConfigureAwait(false);
+            contentStream = await content.ReadAsStreamAsync().ConfigureAwait(false);
+        }
+        catch (Exception exception)
+        {
+            _options?.LogDebug($"Unable to read GraphQL content stream: {exception.Message}");
+            return null;
+        }
+
+        if (!contentStream.CanRead)
+        {
+            return null;
+        }
+
+        var originalPosition = (contentStream.CanSeek) ? contentStream.Position : 0;
+        try
+        {
+            TrySeek(contentStream, 0);
+            // https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/IO/StreamReader.cs#L186
+            // Default parameters other than 'leaveOpen'
+            using var reader = new StreamReader(contentStream, Encoding.UTF8, true,
+                1024, leaveOpen: true);
```

Comment:
I copied this from [`DefaultRequestPayloadExtractor`](https://github.com/getsentry/sentry-dotnet/blob/d4dda091cd4bc957d2b5f582fd3b3e83f9b157ab/src/Sentry/Extensibility/DefaultRequestPayloadExtractor.cs#L24).

We could maybe pull it out to a StreamPayloadExtractor helper that both classes use? That might be a good idea in fact, since I've already improved it by adding the call to `LoadIntoBufferAsync`.

Reply:
Ah, in fact that doesn't work. `LoadIntoBufferAsync` only exists on the `HttpContent` type (and the `DefaultRequestPayloadExtractor` is operating on the `Request.Body`, which is of type `Stream?`). 

It might be best to leave them separate as `GraphQlContentExtractor` can then return an async Task (and hopefully save some memory rather than read the whole string onto the stack before returning). I'll update that now.

Reply:
To answer your original question though, I'm not sure why 1024. The default buffer size for StreamReader is 128 so we could just go with that. We have no way of knowing how big people's GraphQL requests are going to be or what an appropriate buffer size would be, so maybe we just run with the default.